src/GovUk.Education.ExploreEducationStatistics.Content.Services/Cache/PublicationCacheService.cs

```diff
@@ -77,6 +78,32 @@ public Task<IList<PublicationTreeThemeViewModel>> UpdatePublicationTree()
         return _publicationService.GetPublicationTree();
     }
 
+    /// <summary>
+    /// Invalidate all Publications in the cache for the specified Theme
+    /// </summary>
+    public Task<Either<ActionResult, Unit>> InvalidatePublicationsByTheme(Guid themeId)
+    {
+        return _publicationService
+            .ListPublications(themeId: themeId, pageSize: int.MaxValue)
```

Comment:
I don't think `ListPublications` is really designed to be used like this, it's intended to be the endpoint for the public Find Stats search.

It's paginated, sorted, it builds a complex view model that's not used at all here, and internally it produces a SQL query joined with a free text view even if no search term is provided.

It doesn't return superseded/archived publications so invalidating these would be missed.

These publications are still accessible in the public site even though they don't appear in search results.

I would expect that 'invalidating publications by theme' would invalidate all cached publications for the specified theme.